.github/workflows/v2.yaml

```diff
@@ -69,228 +76,52 @@ jobs:
-          .
-      - name: Login to Docker Hub
-        if: github.ref == 'refs/heads/main'
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_PASSWORD }}
-      - name: Push tag(s) to Docker Hub
-        if: github.ref == 'refs/heads/main'
-        run: |
-          docker push composer/composer:2.5
-          docker push composer/composer:2.5-bin
-          docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')
-          docker push composer/composer:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')-bin
-      - name: Login to Amazon Public ECR
-        if: github.ref == 'refs/heads/main'
-        uses: docker/login-action@v1
-        with:
-          registry: public.ecr.aws
-          username: ${{ secrets.AWS_ECR_ACCESS_KEY }}
-          password: ${{ secrets.AWS_ECR_SECRET_KEY }}
-      - name: Push tag(s) to Amazon Public ECR
-        if: github.ref == 'refs/heads/main'
-        run: |
-          docker tag composer/composer:2.5 $ECR_REPO:2.5
-          docker tag composer/composer:2.5-bin $ECR_REPO:2.5-bin
-          docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')
-          docker tag composer/composer:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+') $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')-bin
-          docker push $ECR_REPO:2.5
-          docker push $ECR_REPO:2.5-bin
-          docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')
-          docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION 2.5.\d+' Dockerfile | grep -oP '2.5.\d+')-bin
-  build26:
-    name: Build v2.6
+        docker push $ECR_REPO:${{ matrix.version }}
+        docker push $ECR_REPO:${{ matrix.version }}-bin
+        docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION ${{ matrix.version }}.\d+' Dockerfile | grep -oP '${{ matrix.version }}.\d+')
+        docker push $ECR_REPO:$(grep -oP 'COMPOSER_VERSION ${{ matrix.version }}.\d+' Dockerfile | grep -oP '${{ matrix.version }}.\d+')-bin
+
+  build-latest:
+    name: Build latest version
     runs-on: ubuntu-latest
     timeout-minutes: 20
-    defaults:
-      run:
-        working-directory: '2.6'
+    strategy:
+      matrix:
+        version:
+          - "2.6"
     steps:
       - uses: actions/checkout@v3
       # Build full image: binary with runtime
       - name: Build full image
+        working-directory: ${{ matrix.version }}
         run: |
           docker build \
           --pull \
           --no-cache \
           --target binary-with-runtime \
           --tag composer/composer:latest \
           --tag composer/composer:2 \
```

Comment:
Unless I am mistaken, the only difference between the proposed `build-previous` and `build-latest` version is that we use a tag `2` in addition to `2.x`. Is that right?

Reply:
`latest` and `major` tag version

Reply:
Oh, and `2.2` is also tagged as `lts` to make it a bit more complicated